examples/opt_serving/interactive_hosted.py

```diff
@@ -247,6 +397,12 @@ def index():
     parser.add_argument("--model", type=str, default="alpa/opt-125m")
     parser.add_argument("--path", type=str, default="/home/ubuntu/opt_weights/")
     parser.add_argument("--port", type=int, default=DEFAULT_PORT)
+    parser.add_argument("--serve-logprobs", action="store_true", default=False)
     args = parser.parse_args()
 
+    if args.serve_logprobs:
+        logprobs.provide_automatic_options = False
+        logprobs.methods = ['POST']
```

Comment:
How do you plan to query the endpoint? Do you have client code? Would also be good to share.

Also, we can setup this on AWS and use your client code to do a few tests.

After things are tested stable, we can discuss a time window (say, some weekend 1 - 5am) when your program can run with full capacity

Reply:
Here's some simple example code I've been using to test:

```
import requests
import json

from scipy.special import softmax
import numpy as np
from transformers import AutoTokenizer

tokenizer = AutoTokenizer.from_pretrained('facebook/opt-125m')

url = "http://localhost:9054/logprobs"
headers = {'Content-type': 'application/json', 'Accept': 'text/plain'}
data = {
    "prompt":[int(x) for x in tokenizer.encode("mary had a little")],
    "max_tokens":"64",
    "temperature":"0.7",
    "top_p":"0.5",
    "top_k":"100",
}

r = requests.post(url, data=json.dumps(data), headers=headers)
# import pdb; pdb.set_trace()

for i in range(20):
    output = r.json()
    data['cache_id'] = output['cache_id']
    distribution = np.array([-1e8 for _ in range(tokenizer.vocab_size)])
    print(len(output['logprobs']))
    for idx, logprob in zip(output['indices'], output['logprobs']):
        distribution[idx] = logprob
    distribution = softmax(distribution)
    token = np.random.choice(np.arange(len(distribution)), p=distribution)
    data['prompt'].append(int(token))
    print(tokenizer.decode(data['prompt']))
    r = requests.post(url, data=json.dumps(data), headers=headers)

import pdb; pdb.set_trace()
```

After things are tested + stable, I was hoping to be able to query the endpoint at any time, and just give priority to the completions endpoint when there's traffic (as I implemented with the `made_model_call` logic). Would that be possible?